<h1><b> CLUSTERING-CITRA-SATELIT </b></h1>

📝 **Notebook: 1_preprocessing_data.ipynb**

Notebook ini digunakan untuk melakukan **preprocessing data** sebelum analisis clustering citra satelit bangunan sekolah untuk **14.927 SMA di Indonesia**.  

---

📂 **Sumber Data**

Data input yang digunakan berasal dari tahap sebelumnya, yaitu:  
1. **SCRAPING-DATA-SEKOLAH** → berisi informasi nama sekolah, kabupaten/kota, dan provinsi.  
2. **SCRAPING-CITRA-SATELIT** → berisi citra satelit sekolah (gambar bangunan sekolah).  

Dataset gabungan tersedia dalam file:  

📄 **`../data/data_scraping_maps.csv`**  
File ini diperoleh dari proses **scraping maps** pada proyek **SCRAPING-CITRA-SATELIT**.

---

⚙️ **Tahapan EDA**  

Beberapa langkah eksplorasi data yang dilakukan di notebook ini:  

1. **LIBRARY**  
   - Import library yang dibutuhkan untuk analisis data dan visualisasi (`pandas`, `numpy`, `matplotlib`, `seaborn`, dll).  

2. **CONFIGURATION**  
   - Menyiapkan direktori untuk metadata dan output hasil analisis.  
   - Mengatur style visualisasi dan parameter plotting.  

3. **LOAD METADATA**  
   - Memuat dataset hasil preprocessing dari file `../data/data_preprocessing.csv`.  
   - Mengecek jumlah data, kolom, dan informasi dasar dataset.  

4. **EXPLORATORY DATA ANALYSIS (EDA)**  
   - **Statistika Deskriptif** → Ringkasan data numerik (longitude, latitude, jumlah sekolah).  
   - **Analisis Data Kategorikal** → Distribusi akreditasi sekolah (A, B, C), distribusi per provinsi.  
   - **Analisis Data Geografis** → Visualisasi sebaran sekolah pada peta koordinat, termasuk berdasarkan akreditasi.  
   - **Menyimpan hasil** → Plot/gambar ke direktori `../hasil/eda/` dalam format `.png` atau `.html` untuk dokumentasi dan analisis lebih lanjut.  

---


📊 **Hasil yang Diperoleh**  

Proses preprocessing akan menghasilkan:  

- Dataset sekolah yang **siap digunakan untuk ekstraksi fitur dan clustering**  
- Validasi path citra satelit (hanya menyisakan data dengan status **valid**)  
- Metadata preprocessing dalam bentuk CSV untuk tahap selanjutnya  

Output utama akan disimpan di folder:  
📂 **`../data/data_preprocessing.csv`**  

---


---
# **LIBRARY**
---

In [111]:
# ============================
# Standard Library
# ============================
import os               # operasi terkait file system (path, direktori, file)

# ============================
# Third-Party Libraries
# ============================
import pandas as pd     # manipulasi dan analisis data dalam bentuk tabel (DataFrame)
pd.set_option('display.max_colwidth', None)
import numpy as np      # komputasi numerik dan operasi array/matriks


---
# **CONFIGURATION**
---

In [112]:
# ============================
# Path Configuration
# ============================

# Input dataset hasil scraping
METADATA_PATH = "../data/data_scraping_maps.csv"

# Output directory untuk hasil preprocessing
OUTPUT_DIR = "../data/"

---
# **LOAD METADATA**
---

In [113]:
df_ori = pd.read_csv(METADATA_PATH, sep=",")
df_ori.head()

nomor kode_sekolah                  nama_sekolah  \
0      1     10100139  SMA NEGERI 1 LEMBAH SEULAWAH   
1      2     10100170       SMA NEGERI MODAL BANGSA   
2      3     10100176            SMAS MALEM PUTRA 1   
3      4     10100177          SMA NEGERI 1 LEUPUNG   
4      5     10100179        SMA NEGERI 1 PULO ACEH   

                                                                                  alamat  \
0  JL. BANDA ACEH-MEDAN KM.57,5 Lamtamot Kec. Lembah Seulawah Kab. Aceh Besar Prov. Aceh   
1   JL. BANDARA SIM KM. 12,5 Cot Geundreut Kec. Blang Bintang Kab. Aceh Besar Prov. Aceh   
2            JL.SYEH ABDULLAH KAN-AN Leugeu Kec. Darul Imarah Kab. Aceh Besar Prov. Aceh   
3      JL. BANDA ACEH CALANG KM-18 DAYAH MAMPLAM Kec. Leupung Kab. Aceh Besar Prov. Aceh   
4                JL. RAJA KANDANG Alue Riyeung Kec. Pulo Aceh Kab. Aceh Besar Prov. Aceh   

    kabupaten kota provinsi akreditasi       long       lat  \
0  Aceh Besar  NaN     Aceh          A  95.682881  5.370129   
1  Aceh Besar  NaN     Aceh          A  95.395695  5.508437   
2  Aceh Besar  NaN     Aceh          C  95.330857  5.493038   
3  Aceh Besar  NaN     Aceh          B  95.241723  5.435694   
4  Aceh Besar  NaN     Aceh          B  95.144949  5.615454   

                                                             dir_gambar_ori  \
0  ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                             dir_gambar_titik  \
0  ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                 dir_gambar_kotak_dan_yolo  \
0  ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/smp_malem_putra_1.jpg   
3                ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sman_1_leupung.jpg   
4               ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_1_pulo_aceh.jpg   

  keterangan  
0     sukses  
1     sukses  
2     sukses  
3     sukses  
4     sukses

In [114]:
df_ori.shape

(14927, 14)

Data telah berhasil dimuat dengan rincian sebagai berikut:
- Jumlah baris: 14.927 
- Jumlah kolom: 14

Selanjutnya akan dilakukan preprocessing data.

---
# **PREPROCESSING**
---

Pada tahap ini, dataset asli disalin menjadi sebuah dataframe baru bernama `df_preprocessing_0`.  
Tujuannya adalah agar proses preprocessing dapat dilakukan tanpa mengubah data mentah (*raw data*), sehingga data asli tetap terjaga dan dapat digunakan kembali bila diperlukan.

In [115]:
df_preprocessing_0 = df_ori.copy()

## **Missing Value**

In [116]:
df_preprocessing_0.isnull().sum()

nomor                            0
kode_sekolah                     0
nama_sekolah                     0
alamat                           0
kabupaten                     3598
kota                         11341
provinsi                        12
akreditasi                      25
long                           202
lat                            202
dir_gambar_ori                 202
dir_gambar_titik               202
dir_gambar_kotak_dan_yolo      202
keterangan                       0
dtype: int64

Interpretasi  

- Sebagian besar kolom memiliki data lengkap, seperti **nomor**, **kode_sekolah**, **nama_sekolah**, **link_profil**, dan **alamat**.  
- Kolom **kabupaten** dan **kota** memiliki jumlah *missing values* cukup besar, terutama pada kolom **kota**.  
- Kolom **provinsi** dan **akreditasi** hanya memiliki sedikit *missing values*.  
- Kolom koordinat (**long**, **lat**) serta direktori gambar (**dir_gambar_ori**, **dir_gambar_titik**, **dir_gambar_kotak_dan_yolo**) masing-masing hilang pada 202 baris.  
- Kolom **keterangan** tidak memiliki *missing values*.  

Tahapan preprocessing selanjutnya akan difokuskan pada **penanganan nilai yang hilang** agar dataset lebih bersih dan siap digunakan.  

Untuk melihat lebih lanjut, prosesnya akan dibagi menjadi 4.
1. Kolom **kabupaten** dan **kota** (3598 + 11341 baris)
2. Kolom **provinsi** (12 baris)
3. Kolom **akreditasi** (25 baris)
4. Kolom koordinat **long**, **lat** serta direktori gambar **dir_gambar_ori**, **dir_gambar_titik**, **dir_gambar_kotak_dan_yolo**(Masing-masing 202 baris)


### 1. Kolom `kabupaten` dan `kota`

In [117]:
kondisi = df_preprocessing_0['kabupaten'].isnull()
df_preprocessing_0[kondisi].head()

nomor kode_sekolah                nama_sekolah  \
132    133     10104040   SMA NEGERI 1 SIMPANG KIRI   
133    134     10104041            SMA HIDAYATULLAH   
136    137     10104044  SMA NEGERI 1 SULTAN DAULAT   
141    142     10104059              SMAN 1 RUNDENG   
179    180     10105272               SMAN 2 SABANG   

                                                                                              alamat  \
132    JL. SYEKH ABDURRAUF NO. 1 Subulussalam Selatan Kec. Simpang Kiri Kota Subulussalam Prov. Aceh   
133  Jln. Hamzah Fansyuri No. 86 Subulussalam Selatan Kec. Simpang Kiri Kota Subulussalam Prov. Aceh   
136                     Jl. Darul Makmur No. 1 Jambi Kec. Sultan Daulat Kota Subulussalam Prov. Aceh   
141                   JL. PERJUANGAN NO.14 Muara Batu-Batu Kec. Rundeng Kota Subulussalam Prov. Aceh   
179                                 JL. TINJAU ALAM Aneuk Laot Kec. Sukakarya Kota Sabang Prov. Aceh   

    kabupaten               kota provinsi akreditasi       long       lat  \
132       NaN  Kota Subulussalam     Aceh          A  97.999012  2.645893   
133       NaN  Kota Subulussalam     Aceh          C  97.997368  2.645271   
136       NaN  Kota Subulussalam     Aceh          B  97.901610  2.792601   
141       NaN  Kota Subulussalam     Aceh          B  97.857891  2.669041   
179       NaN        Kota Sabang     Aceh          A  95.329182  5.874299   

                                                                    dir_gambar_ori  \
132       ../hasil/gambar_ori/ACEH/KOTA SUBULUSSALAM/sma_negeri_1_simpang_kiri.jpg   
133  ../hasil/gambar_ori/ACEH/KOTA SUBULUSSALAM/mtss_hidayatullah_subulussalam.jpg   
136      ../hasil/gambar_ori/ACEH/KOTA SUBULUSSALAM/sma_negeri_1_sultan_daulat.jpg   
141            ../hasil/gambar_ori/ACEH/KOTA SUBULUSSALAM/sma_negeri_1_rundeng.jpg   
179                   ../hasil/gambar_ori/ACEH/KOTA SABANG/sma_negeri_2_sabang.jpg   

                                                                    dir_gambar_titik  \
132       ../hasil/gambar_titik/ACEH/KOTA SUBULUSSALAM/sma_negeri_1_simpang_kiri.jpg   
133  ../hasil/gambar_titik/ACEH/KOTA SUBULUSSALAM/mtss_hidayatullah_subulussalam.jpg   
136      ../hasil/gambar_titik/ACEH/KOTA SUBULUSSALAM/sma_negeri_1_sultan_daulat.jpg   
141            ../hasil/gambar_titik/ACEH/KOTA SUBULUSSALAM/sma_negeri_1_rundeng.jpg   
179                   ../hasil/gambar_titik/ACEH/KOTA SABANG/sma_negeri_2_sabang.jpg   

                                                                        dir_gambar_kotak_dan_yolo  \
132       ../hasil/gambar_kotak_dan_yolo/ACEH/KOTA SUBULUSSALAM/img/sma_negeri_1_simpang_kiri.jpg   
133  ../hasil/gambar_kotak_dan_yolo/ACEH/KOTA SUBULUSSALAM/img/mtss_hidayatullah_subulussalam.jpg   
136      ../hasil/gambar_kotak_dan_yolo/ACEH/KOTA SUBULUSSALAM/img/sma_negeri_1_sultan_daulat.jpg   
141            ../hasil/gambar_kotak_dan_yolo/ACEH/KOTA SUBULUSSALAM/img/sma_negeri_1_rundeng.jpg   
179                   ../hasil/gambar_kotak_dan_yolo/ACEH/KOTA SABANG/img/sma_negeri_2_sabang.jpg   

      keterangan  
132       sukses  
133  cek kembali  
136       sukses  
141       sukses  
179       sukses

In [118]:
kondisi = df_preprocessing_0['kota'].isnull()
df_preprocessing_0[kondisi].head()

nomor kode_sekolah                  nama_sekolah  \
0      1     10100139  SMA NEGERI 1 LEMBAH SEULAWAH   
1      2     10100170       SMA NEGERI MODAL BANGSA   
2      3     10100176            SMAS MALEM PUTRA 1   
3      4     10100177          SMA NEGERI 1 LEUPUNG   
4      5     10100179        SMA NEGERI 1 PULO ACEH   

                                                                                  alamat  \
0  JL. BANDA ACEH-MEDAN KM.57,5 Lamtamot Kec. Lembah Seulawah Kab. Aceh Besar Prov. Aceh   
1   JL. BANDARA SIM KM. 12,5 Cot Geundreut Kec. Blang Bintang Kab. Aceh Besar Prov. Aceh   
2            JL.SYEH ABDULLAH KAN-AN Leugeu Kec. Darul Imarah Kab. Aceh Besar Prov. Aceh   
3      JL. BANDA ACEH CALANG KM-18 DAYAH MAMPLAM Kec. Leupung Kab. Aceh Besar Prov. Aceh   
4                JL. RAJA KANDANG Alue Riyeung Kec. Pulo Aceh Kab. Aceh Besar Prov. Aceh   

    kabupaten kota provinsi akreditasi       long       lat  \
0  Aceh Besar  NaN     Aceh          A  95.682881  5.370129   
1  Aceh Besar  NaN     Aceh          A  95.395695  5.508437   
2  Aceh Besar  NaN     Aceh          C  95.330857  5.493038   
3  Aceh Besar  NaN     Aceh          B  95.241723  5.435694   
4  Aceh Besar  NaN     Aceh          B  95.144949  5.615454   

                                                             dir_gambar_ori  \
0  ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                             dir_gambar_titik  \
0  ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                 dir_gambar_kotak_dan_yolo  \
0  ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/smp_malem_putra_1.jpg   
3                ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sman_1_leupung.jpg   
4               ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_1_pulo_aceh.jpg   

  keterangan  
0     sukses  
1     sukses  
2     sukses  
3     sukses  
4     sukses

Dapat dilihat bahwa nilai `null` pada kolom **kabupaten** terjadi ketika sekolah tersebut berada di kota, 
dan sebaliknya nilai `null` pada kolom **kota** muncul ketika sekolah berada di kabupaten.  

Untuk mengatasi hal ini, kedua kolom tersebut akan digabungkan menjadi satu kolom baru bernama **kabupaten_kota**:  
- Jika data berasal dari kota, maka nilainya tetap menggunakan nama kota.  
- Jika data berasal dari kabupaten, maka nama kabupaten akan ditambahkan awalan **"KAB "** agar lebih konsisten dalam format penamaan.  


In [119]:
df_preprocessing_1 = df_preprocessing_0.copy()

data_kabupaten_kota = []

# Iterasi baris demi baris pada dataframe
for _, row in df_preprocessing_1.iterrows():
    # Jika kolom 'kabupaten' kosong, gunakan nilai dari 'kota'
    # Jika ada nilai 'kabupaten', tambahkan prefix "Kab " di depannya
    data_kabupaten_kota.append(
        row['kota'] if pd.isnull(row['kabupaten']) else f"Kab {row['kabupaten']}"
    )

df_preprocessing_1['kabupaten_kota'] = data_kabupaten_kota
df_preprocessing_1.head()


nomor kode_sekolah                  nama_sekolah  \
0      1     10100139  SMA NEGERI 1 LEMBAH SEULAWAH   
1      2     10100170       SMA NEGERI MODAL BANGSA   
2      3     10100176            SMAS MALEM PUTRA 1   
3      4     10100177          SMA NEGERI 1 LEUPUNG   
4      5     10100179        SMA NEGERI 1 PULO ACEH   

                                                                                  alamat  \
0  JL. BANDA ACEH-MEDAN KM.57,5 Lamtamot Kec. Lembah Seulawah Kab. Aceh Besar Prov. Aceh   
1   JL. BANDARA SIM KM. 12,5 Cot Geundreut Kec. Blang Bintang Kab. Aceh Besar Prov. Aceh   
2            JL.SYEH ABDULLAH KAN-AN Leugeu Kec. Darul Imarah Kab. Aceh Besar Prov. Aceh   
3      JL. BANDA ACEH CALANG KM-18 DAYAH MAMPLAM Kec. Leupung Kab. Aceh Besar Prov. Aceh   
4                JL. RAJA KANDANG Alue Riyeung Kec. Pulo Aceh Kab. Aceh Besar Prov. Aceh   

    kabupaten kota provinsi akreditasi       long       lat  \
0  Aceh Besar  NaN     Aceh          A  95.682881  5.370129   
1  Aceh Besar  NaN     Aceh          A  95.395695  5.508437   
2  Aceh Besar  NaN     Aceh          C  95.330857  5.493038   
3  Aceh Besar  NaN     Aceh          B  95.241723  5.435694   
4  Aceh Besar  NaN     Aceh          B  95.144949  5.615454   

                                                             dir_gambar_ori  \
0  ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                             dir_gambar_titik  \
0  ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                 dir_gambar_kotak_dan_yolo  \
0  ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg   
1       ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_modal_bangsa.jpg   
2             ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/smp_malem_putra_1.jpg   
3                ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sman_1_leupung.jpg   
4               ../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_1_pulo_aceh.jpg   

  keterangan  kabupaten_kota  
0     sukses  Kab Aceh Besar  
1     sukses  Kab Aceh Besar  
2     sukses  Kab Aceh Besar  
3     sukses  Kab Aceh Besar  
4     sukses  Kab Aceh Besar

In [120]:
df_preprocessing_1.isnull().sum()

nomor                            0
kode_sekolah                     0
nama_sekolah                     0
alamat                           0
kabupaten                     3598
kota                         11341
provinsi                        12
akreditasi                      25
long                           202
lat                            202
dir_gambar_ori                 202
dir_gambar_titik               202
dir_gambar_kotak_dan_yolo      202
keterangan                       0
kabupaten_kota                  12
dtype: int64

Setelah dilakukan proses penggabungan, masih terdapat **12 data** pada kolom kabupaten_kota yang bernilai kosong.

In [121]:
kondisi = df_preprocessing_1['kabupaten_kota'].isnull()
df_preprocessing_1[kondisi]

nomor kode_sekolah                                 nama_sekolah  \
14912  14913     90101003               SMA Sekolah Indonesia Den Haag   
14913  14914     90101006                 SEKOLAH SI CAIRO TINGKAT SMA   
14914  14915     90101009         SEKOLAH INDONESIA RIYADH TINGKAT SMA   
14915  14916     90101016         SEKOLAH INDONESIA JEDDAH TINGKAT SMA   
14916  14917     90101019          SEKOLAH SI KUALA LUMPUR TINGKAT SMA   
14917  14918     90101022    SEKOLAH INDONESIA (SINGAPURA) TINGKAT SMA   
14918  14919     90101025                 Sekolah SI Tokyo Tingkat SMA   
14919  14920     90101028               SEKOLAH SI BANGKOK TINGKAT SMA   
14920  14921     90101031                SEKOLAH SI YANGON TINGKAT SMA   
14921  14922     90101034                 SEKOLAH SI DAVAO TINGKAT SMA   
14922  14923     90101053  SEKOLAH INDONESIA KOTA KINABALU TINGKAT SMA   
14923  14924     90101190         SEKOLAH INDONESIA MAKKAH TINGKAT SMA   

                                                                                               alamat  \
14912             Rijksstraatweg 679, 2245 CB Wassenaar, T... Wassenaar Wassenaar Belanda Luar Negeri   
14913                       63 MUSADDAK STR. DOKKI , PO. BOX 1661, C... Cairo Cairo Mesir Luar Negeri   
14914                8561 Abdullah Ibn Muammar, Umm Al Hamam ... Riyadh Riyadh Arab Saudi Luar Negeri   
14915                Districk 7155,5421 Az Zilaq, Ar Rehab, J... Jeddah Jeddah Arab Saudi Luar Negeri   
14916      No. 1 Lorong Tun Ismail 50480, Kuala Lum... Kuala Lumpur Kuala Lumpur Malaysia Luar Negeri   
14917                SINGAPURA SIGLAP ROAD SINGAPORE 455859 Singapura Singapura Singapura Luar Negeri   
14918                   Sekolah Indonesia Tokyo No. 4-6-6, Megur... Meguro-ku Tokyo Japan Luar Negeri   
14919              Indonesian Embassy 600-602 Petchburi Roa... Phayathai Bangkok Thailand Luar Negeri   
14920                   100, LOWER KYIMYINDINE ROAD, AHLONE YANG... Yangon Yangon Myanmar Luar Negeri   
14921          KONSULAT JENDERAL RI ECOLAND SUBDIVISION... Davao City Davao City Filipina Luar Negeri   
14922         Jalan 3B No 6 KKIP Selatan Dua Kota Kina... Kinabalu Kota Kinabalu Malaysia Luar Negeri   
14923  Hera Complex, District Al-Gassalah Jabal... District Al-Gassalah Mekkah Arab Saudi Luar Negeri   

      kabupaten kota provinsi akreditasi        long        lat  \
14912       NaN  NaN      NaN          A    4.374131  52.114105   
14913       NaN  NaN      NaN          A   31.258464  30.059489   
14914       NaN  NaN      NaN          A   46.644902  24.697719   
14915       NaN  NaN      NaN          A   39.234882  21.545279   
14916       NaN  NaN      NaN          A  101.690729   3.164676   
14917       NaN  NaN      NaN          A  103.926097   1.310750   
14918       NaN  NaN      NaN          A  110.460236  -7.006764   
14919       NaN  NaN      NaN          A  100.535955  13.750365   
14920       NaN  NaN      NaN          A   96.133895  16.782880   
14921       NaN  NaN      NaN          A  125.597527   7.049427   
14922       NaN  NaN      NaN          A  116.152390   6.079791   
14923       NaN  NaN      NaN          B   39.864902  21.443535   

                                                                                                         dir_gambar_ori  \
14912                                                            ../hasil/gambar_ori/indonesian_school_of_the_hague.jpg   
14913                                                                                     ../hasil/gambar_ori/kairo.jpg   
14914                                                 ../hasil/gambar_ori/(new)_indonesian_embassy_school_in_riyadh.jpg   
14915                               ../hasil/gambar_ori/indonesian_embassy_school_jeddah_-_sekolah_indonesia_jeddah.jpg   
14916                                                            ../hasil/gambar_ori/sekolah_indonesia_kuala_lumpur.jpg   
14917                                                               

Setelah dicek, ternyata **12 data** yang kosong pada kolom kabupaten_kota juga merupakan data yang kosong pada kolom `provinsi`. <br>
Oleh karena itu, perbaikan terhadap data ini akan ditindaklanjuti bersamaan dengan penanganan pada kolom `provinsi`.

### 2. Kolom `provinsi`

In [122]:
kondisi = df_preprocessing_1['provinsi'].isnull()
df_preprocessing_1[kondisi]

nomor kode_sekolah                                 nama_sekolah  \
14912  14913     90101003               SMA Sekolah Indonesia Den Haag   
14913  14914     90101006                 SEKOLAH SI CAIRO TINGKAT SMA   
14914  14915     90101009         SEKOLAH INDONESIA RIYADH TINGKAT SMA   
14915  14916     90101016         SEKOLAH INDONESIA JEDDAH TINGKAT SMA   
14916  14917     90101019          SEKOLAH SI KUALA LUMPUR TINGKAT SMA   
14917  14918     90101022    SEKOLAH INDONESIA (SINGAPURA) TINGKAT SMA   
14918  14919     90101025                 Sekolah SI Tokyo Tingkat SMA   
14919  14920     90101028               SEKOLAH SI BANGKOK TINGKAT SMA   
14920  14921     90101031                SEKOLAH SI YANGON TINGKAT SMA   
14921  14922     90101034                 SEKOLAH SI DAVAO TINGKAT SMA   
14922  14923     90101053  SEKOLAH INDONESIA KOTA KINABALU TINGKAT SMA   
14923  14924     90101190         SEKOLAH INDONESIA MAKKAH TINGKAT SMA   

                                                                                               alamat  \
14912             Rijksstraatweg 679, 2245 CB Wassenaar, T... Wassenaar Wassenaar Belanda Luar Negeri   
14913                       63 MUSADDAK STR. DOKKI , PO. BOX 1661, C... Cairo Cairo Mesir Luar Negeri   
14914                8561 Abdullah Ibn Muammar, Umm Al Hamam ... Riyadh Riyadh Arab Saudi Luar Negeri   
14915                Districk 7155,5421 Az Zilaq, Ar Rehab, J... Jeddah Jeddah Arab Saudi Luar Negeri   
14916      No. 1 Lorong Tun Ismail 50480, Kuala Lum... Kuala Lumpur Kuala Lumpur Malaysia Luar Negeri   
14917                SINGAPURA SIGLAP ROAD SINGAPORE 455859 Singapura Singapura Singapura Luar Negeri   
14918                   Sekolah Indonesia Tokyo No. 4-6-6, Megur... Meguro-ku Tokyo Japan Luar Negeri   
14919              Indonesian Embassy 600-602 Petchburi Roa... Phayathai Bangkok Thailand Luar Negeri   
14920                   100, LOWER KYIMYINDINE ROAD, AHLONE YANG... Yangon Yangon Myanmar Luar Negeri   
14921          KONSULAT JENDERAL RI ECOLAND SUBDIVISION... Davao City Davao City Filipina Luar Negeri   
14922         Jalan 3B No 6 KKIP Selatan Dua Kota Kina... Kinabalu Kota Kinabalu Malaysia Luar Negeri   
14923  Hera Complex, District Al-Gassalah Jabal... District Al-Gassalah Mekkah Arab Saudi Luar Negeri   

      kabupaten kota provinsi akreditasi        long        lat  \
14912       NaN  NaN      NaN          A    4.374131  52.114105   
14913       NaN  NaN      NaN          A   31.258464  30.059489   
14914       NaN  NaN      NaN          A   46.644902  24.697719   
14915       NaN  NaN      NaN          A   39.234882  21.545279   
14916       NaN  NaN      NaN          A  101.690729   3.164676   
14917       NaN  NaN      NaN          A  103.926097   1.310750   
14918       NaN  NaN      NaN          A  110.460236  -7.006764   
14919       NaN  NaN      NaN          A  100.535955  13.750365   
14920       NaN  NaN      NaN          A   96.133895  16.782880   
14921       NaN  NaN      NaN          A  125.597527   7.049427   
14922       NaN  NaN      NaN          A  116.152390   6.079791   
14923       NaN  NaN      NaN          B   39.864902  21.443535   

                                                                                                         dir_gambar_ori  \
14912                                                            ../hasil/gambar_ori/indonesian_school_of_the_hague.jpg   
14913                                                                                     ../hasil/gambar_ori/kairo.jpg   
14914                                                 ../hasil/gambar_ori/(new)_indonesian_embassy_school_in_riyadh.jpg   
14915                               ../hasil/gambar_ori/indonesian_embassy_school_jeddah_-_sekolah_indonesia_jeddah.jpg   
14916                                                            ../hasil/gambar_ori/sekolah_indonesia_kuala_lumpur.jpg   
14917                                                               

Dapat dilihat bahwa **12 data** tersebut kosong pada kolom `kabupaten`, `kota`, maupun `provinsi`.  
Berdasarkan alamat dan koordinatnya, lokasi sekolah tersebut **tidak berada di Indonesia**.  

Dengan demikian, **12 sekolah** ini merupakan **sekolah Indonesia yang berlokasi di luar negeri**, sehingga akan **dihapus (drop)** dari dataset.


In [123]:
# Menghapus 12 sekolah yang berada di luar negeri (tanpa kabupaten, kota, dan provinsi)
df_preprocessing_2 = df_preprocessing_1.dropna(subset=["provinsi"])

print("Jumlah data sebelum drop:", len(df_preprocessing_1))
print("Jumlah data setelah drop:", len(df_preprocessing_2))

kondisi = df_preprocessing_2[df_preprocessing_2['provinsi'].isnull()]
df_preprocessing_2.isnull().sum()

Jumlah data sebelum drop: 14927
Jumlah data setelah drop: 14915


nomor                            0
kode_sekolah                     0
nama_sekolah                     0
alamat                           0
kabupaten                     3586
kota                         11329
provinsi                         0
akreditasi                      25
long                           202
lat                            202
dir_gambar_ori                 202
dir_gambar_titik               202
dir_gambar_kotak_dan_yolo      202
keterangan                       0
kabupaten_kota                   0
dtype: int64

### 3. Kolom `akreditasi`

In [124]:
kondisi = df_preprocessing_2['akreditasi'].isnull()
df_preprocessing_2[kondisi].head()

nomor kode_sekolah           nama_sekolah  \
1828   1829     10495729       SMAS NUR HASANAH   
3539   3540     20107367  SMAS AL JIHAD JAKARTA   
3954   3955     20219723               SMAS BPN   
3981   3982     20219770     SMAS PAHLAWAN TOHA   
4911   4912     20306193      SMAS DARUL HIKMAH   

                                                                                            alamat  \
1828         JL. LINTAS TANJUNG MEDAN BAKTI MAKMUR Kec. Bagan Sinembah Kab. Rokan Hilir Prov. Riau   
3539  JL. JIHAD PAPANGGO I Papango Kec. Tanjung Priok Kota Adm. Jakarta Utara Prov. D.K.I. Jakarta   
3954                    JL.DESA CIPADUNG NO.213 Cipadung Kec. Cibiru Kota Bandung Prov. Jawa Barat   
3981          Jl. Peta (Sukamulya I) Suka Asih Kec. Bojong Loa Kaler Kota Bandung Prov. Jawa Barat   
4911               JL. S. PARMAN, KUTOARJO Kutoarjo Kec. Kutoarjo Kab. Purworejo Prov. Jawa Tengah   

        kabupaten                     kota        provinsi akreditasi  \
1828  Rokan Hilir                      NaN            Riau        NaN   
3539          NaN  Kota Adm. Jakarta Utara  D.K.I. Jakarta        NaN   
3954          NaN             Kota Bandung      Jawa Barat        NaN   
3981          NaN             Kota Bandung      Jawa Barat        NaN   
4911    Purworejo                      NaN     Jawa Tengah        NaN   

            long       lat  \
1828   98.701055  3.547577   
3539  106.882097 -6.130116   
3954  107.717654 -6.926360   
3981  107.591123 -6.935669   
4911  109.910789 -7.716324   

                                                                                        dir_gambar_ori  \
1828              ../hasil/gambar_ori/RIAU/KAB ROKAN HILIR/sekolah_menengah_kejuruan_i_nur_hasanah.jpg   
3539  ../hasil/gambar_ori/D.K.I. JAKARTA/KOTA ADM. JAKARTA UTARA/madrasah-madrasah_aliyah_al-jihad.jpg   
3954                                          ../hasil/gambar_ori/JAWA BARAT/KOTA BANDUNG/smas_bpn.jpg   
3981                        ../hasil/gambar_ori/JAWA BARAT/KOTA BANDUNG/smas_pahlawan_toha_bandung.jpg   
4911            ../hasil/gambar_ori/JAWA TENGAH/KAB PURWOREJO/darul_hikmah_kutoarjo_islamic_school.jpg   

                                                                                        dir_gambar_titik  \
1828              ../hasil/gambar_titik/RIAU/KAB ROKAN HILIR/sekolah_menengah_kejuruan_i_nur_hasanah.jpg   
3539  ../hasil/gambar_titik/D.K.I. JAKARTA/KOTA ADM. JAKARTA UTARA/madrasah-madrasah_aliyah_al-jihad.jpg   
3954                                          ../hasil/gambar_titik/JAWA BARAT/KOTA BANDUNG/smas_bpn.jpg   
3981                        ../hasil/gambar_titik/JAWA BARAT/KOTA BANDUNG/smas_pahlawan_toha_bandung.jpg   
4911            ../hasil/gambar_titik/JAWA TENGAH/KAB PURWOREJO/darul_hikmah_kutoarjo_islamic_school.jpg   

                                                                                            dir_gambar_kotak_dan_yolo  \
1828              ../hasil/gambar_kotak_dan_yolo/RIAU/KAB ROKAN HILIR/img/sekolah_menengah_kejuruan_i_nur_hasanah.jpg   
3539  ../hasil/gambar_kotak_dan_yolo/D.K.I. JAKARTA/KOTA ADM. JAKARTA UTARA/img/madrasah-madrasah_aliyah_al-jihad.jpg   
3954                                          ../hasil/gambar_kotak_dan_yolo/JAWA BARAT/KOTA BANDUNG/img/smas_bpn.jpg   
3981                        ../hasil/gambar_kotak_dan_yolo/JAWA BARAT/KOTA BANDUNG/img/smas_pahlawan_toha_bandung.jpg   
4911            ../hasil/gambar_kotak_dan_yolo/JAWA TENGAH/KAB PURWOREJO/img/darul_hikmah_kutoarjo_islamic_school.jpg   

       keterangan           kabupaten_kota  
1828       sukses          Kab Rokan Hilir  
3539  cek kembali  Kota Adm. Jakarta Utara  
3954       sukses             Kota Bandung  
3981       sukses             Kota Bandung  
4911       sukses            Kab Purworejo

Terdapat **25 data** yang kosong pada kolom `akreditasi`.  
Selain itu, kita juga perlu mengetahui **nilai** apa saja yang terdapat pada kolom `akreditasi` untuk memahami distribusi kategorinya.

In [125]:
df_preprocessing_2['akreditasi'].value_counts()

akreditasi
A                      6785
B                      5288
C                      2189
-                       568
Tidak Terakreditasi      60
Name: count, dtype: int64

Berdasarkan hasil eksplorasi, terdapat **5 kategori akreditasi** pada data, yaitu:  
- **A** : 6.785 sekolah  
- **B** : 5.288 sekolah  
- **C** : 2.189 sekolah  
- **-** : 568 sekolah  
- **Tidak Terakreditasi** : 60 sekolah  

Selain itu, masih ada **25 data kosong (null)** pada kolom `akreditasi`.  

Karena tujuan penelitian ini adalah membuat **sistem analisis kelayakan** yang dapat dievaluasi, maka kategori **"-"**, **"Tidak Terakreditasi"**, dan **nilai kosong** akan menyulitkan proses analisis.  
Oleh karena itu, sekolah dengan kategori tersebut **tidak akan disertakan dalam analisis**.  

Catatan: <br>
Jumlah data yang akan dihapus (kategori "-", "Tidak Terakreditasi", dan nilai kosong) adalah **653 sekolah**, atau sekitar **4,37%** dari total keseluruhan data (14.927 sekolah).  
Dengan persentase yang relatif kecil ini, penghapusan data tersebut **tidak akan memberikan dampak signifikan** terhadap hasil analisis.



In [126]:
# Daftar akreditasi yang ingin dihapus
hapus_akreditasi = ["-", "Tidak Terakreditasi"]

# Menghapus baris dengan akreditasi "-", "Tidak Terakreditasi", atau null
df_preprocessing_3 = df_preprocessing_2[~df_preprocessing_2["akreditasi"].isin(hapus_akreditasi)]
df_preprocessing_3 = df_preprocessing_3.dropna(subset=["akreditasi"]).reset_index(drop=True)

# Jumlah data setelah filter
print("Jumlah data sebelum drop:", len(df_preprocessing_2))
print("Jumlah data setelah drop:", len(df_preprocessing_3))

# Distribusi akreditasi yang tersisa
print(df_preprocessing_3["akreditasi"].value_counts())
df_preprocessing_3.isnull().sum()

Jumlah data sebelum drop: 14915
Jumlah data setelah drop: 14262
akreditasi
A    6785
B    5288
C    2189
Name: count, dtype: int64


nomor                            0
kode_sekolah                     0
nama_sekolah                     0
alamat                           0
kabupaten                     3420
kota                         10842
provinsi                         0
akreditasi                       0
long                           186
lat                            186
dir_gambar_ori                 186
dir_gambar_titik               186
dir_gambar_kotak_dan_yolo      186
keterangan                       0
kabupaten_kota                   0
dtype: int64

### 4. Kolom koordinat `long`, `lat` serta direktori gambar `dir_gambar_ori`, `dir_gambar_titik`, `dir_gambar_kota_dan_yolo`

In [127]:
kondisi = df_preprocessing_3['long'].isnull()
df_preprocessing_3[kondisi]

nomor kode_sekolah                       nama_sekolah  \
1569    1573     10400846                      SMAN 2 BANTAN   
1577    1581     10400857                      SMAN 1 BANTAN   
1583    1587     10400868      SMA NEGERI 1 TASIK PUTRI PUYU   
1593    1597     10401512                   SMAS PGRI RENGAT   
1597    1601     10401519                      SMAN 2 RENGAT   
...      ...          ...                                ...   
13851  13933     70029996                   SMA IT SOEMAN HS   
13853  13935     70030131                SMA IT KAHAR RAHMAN   
14040  14137     70038448  SMAS GLOBAL SAINS HALMAHERA TIMUR   
14142  14290     70042240         SMA NEGERI 3 ROKAN IV KOTO   
14157  14315     70042824            SMA ISLAM AN NUR SANDAI   

                                                                                                      alamat  \
1569                                      JL. BUDI LUHUR Desa Sukamaju Kec. Bantan Kab. Bengkalis Prov. Riau   
1577                                     JL. SOEKARNO-HATTA Selat Baru Kec. Bantan Kab. Bengkalis Prov. Riau   
1583                 Jl. H. Tamrin no 01 kudap Kudap Kec. Tasik Putri Puyu Kab. Kepulauan Meranti Prov. Riau   
1593                       JL. SULTAN KM. 3 RENGAT Kampung Dagang Kec. Rengat Kab. Indragiri Hulu Prov. Riau   
1597                   JL. HANG TUAH KM. 3 RENGAT Sungai Beringin Kec. Rengat Kab. Indragiri Hulu Prov. Riau   
...                                                                                                      ...   
13851                           Jalan Gunung Raya No.43 Rejosari Kec. Tenayan Raya Kota Pekanbaru Prov. Riau   
13853                  JL.AZKI ARIS RT.015/ RW.003 Kampung Dagang Kec. Rengat Kab. Indragiri Hulu Prov. Riau   
14040                         Desa Lolosita Lolasita Kec. Maba Utara Kab. Halmahera Timur Prov. Maluku Utara   
14142  Jl. Kabupaten Banjar Datar - Tandikat Cipang Kiri Hilir Kec. Rokan IV Koto Kab. Rokan Hulu Prov. Riau   
14157                        Jl. Bina Desa Indralaya Sandai Kec. Sandai Kab. Ketapang Prov. Kalimantan Barat   

               kabupaten            kota          provinsi akreditasi  long  \
1569           Bengkalis             NaN              Riau          A   NaN   
1577           Bengkalis             NaN              Riau          A   NaN   
1583   Kepulauan Meranti             NaN              Riau          B   NaN   
1593      Indragiri Hulu             NaN              Riau          B   NaN   
1597      Indragiri Hulu             NaN              Riau          A   NaN   
...                  ...             ...               ...        ...   ...   
13851                NaN  Kota Pekanbaru              Riau          A   NaN   
13853     Indragiri Hulu             NaN              Riau          B   NaN   
14040    Halmahera Timur             NaN      Maluku Utara          C   NaN   
14142         Rokan Hulu             NaN              Riau          C   NaN   
14157           Ketapang             NaN  Kalimantan Barat          C   NaN   

       lat dir_gambar_ori dir_gambar_titik dir_gambar_kotak_dan_yolo  \
1569   NaN            NaN              NaN                       NaN   
1577   NaN            NaN              NaN                       NaN   
1583   NaN            NaN              NaN                       NaN   
1593   NaN            NaN              NaN                       NaN   
1597   NaN            NaN              NaN                       NaN   
...    ...            ...              ...                       ...   
13851  NaN            NaN              NaN                       NaN   
13853  NaN            NaN              NaN                       NaN   
14040  NaN            NaN              NaN                       NaN   
14142  NaN            NaN              NaN                       NaN   
14157  NaN            NaN              NaN                       NaN   

            keterangan         kabupaten_kota  
1569   tidak ditemukan         

In [128]:
df_preprocessing_3['keterangan'].value_counts()

keterangan
sukses             13187
cek kembali          889
tidak ditemukan      186
Name: count, dtype: int64

Terdapat **186 data kosong** pada kolom `long`, `lat`, `dir_gambar_ori`, `dir_gambar_titik`, dan `dir_gambar_kotak_dan_yolo`.  

Diketahui bahwa data ini merupakan hasil **gagal scraping citra satelit** pada tahap sebelumnya.  
Karena nilai-nilai ini tidak dapat diperbaiki, maka **186 baris ini akan dihapus (drop)** dari dataset.

In [129]:
# Kolom yang harus dicek untuk missing values
kolom_gambar = ["long", "lat", "dir_gambar_ori", "dir_gambar_titik", "dir_gambar_kotak_dan_yolo"]

# Menghapus baris yang memiliki missing values di salah satu kolom tersebut
df_preprocessing_4 = df_preprocessing_3.dropna(subset=kolom_gambar).reset_index(drop=True)

# Jumlah data setelah penghapusan
print("Jumlah data sebelum menghapus baris dengan gambar/koordinat kosong:", len(df_preprocessing_3))
print("Jumlah data setelah menghapus baris dengan gambar/koordinat kosong:", len(df_preprocessing_4))

df_preprocessing_4.isnull().sum()

Jumlah data sebelum menghapus baris dengan gambar/koordinat kosong: 14262
Jumlah data setelah menghapus baris dengan gambar/koordinat kosong: 14076


nomor                            0
kode_sekolah                     0
nama_sekolah                     0
alamat                           0
kabupaten                     3413
kota                         10663
provinsi                         0
akreditasi                       0
long                             0
lat                              0
dir_gambar_ori                   0
dir_gambar_titik                 0
dir_gambar_kotak_dan_yolo        0
keterangan                       0
kabupaten_kota                   0
dtype: int64

Data sekarang sudah **bersih** dari nilai kosong yang bermasalah.  
Kolom **kabupaten** dan **kota** kini **diwakilkan oleh kolom baru** `kabupaten_kota`, sehingga kolom asli akan dihapus jika tidak diperlukan lagi.

In [130]:
# Menghapus kolom kabupaten dan kota karena sudah digantikan oleh kabupaten_kota
df_preprocessing_5 = df_preprocessing_4.drop(columns=["kabupaten", "kota"])
print("Shape dataframe setelah drop kolom:", df_preprocessing_5.shape)

# Cek jumlah missing values di setiap kolom
print("\nJumlah missing values per kolom:")
print(df_preprocessing_5.isnull().sum())


Shape dataframe setelah drop kolom: (14076, 13)

Jumlah missing values per kolom:
nomor                        0
kode_sekolah                 0
nama_sekolah                 0
alamat                       0
provinsi                     0
akreditasi                   0
long                         0
lat                          0
dir_gambar_ori               0
dir_gambar_titik             0
dir_gambar_kotak_dan_yolo    0
keterangan                   0
kabupaten_kota               0
dtype: int64


Karena dataset sekarang sudah bersih dari nilai kosong, tahap selanjutnya adalah melakukan **pengecekan duplikat** pada data.  

## Duplicate Value

In [131]:
df_preprocessing_5.duplicated().sum()

0

Setelah pengecekan duplikat, diketahui bahwa **tidak ada data yang duplikat** pada dataset.  
Tahap selanjutnya adalah melakukan **penyesuaian direktori file gambar** agar path direktori citra sesuai dengan struktur folder saat ini dan memudahkan proses selanjutnya.

## Penyesuaian Direktori

In [132]:
print(df_preprocessing_5['dir_gambar_ori'][0])
print(df_preprocessing_5['dir_gambar_titik'][0])
print(df_preprocessing_5['dir_gambar_kotak_dan_yolo'][0])

../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg


Direktori file gambar saat ini berasal dari proyek sebelumnya **"SCRAPING-CITRA-SATELIT"**, dengan path seperti:  
```
../hasil/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../hasil/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../hasil/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg
```
Untuk kebutuhan proyek **CLUSTERING-CITRA-SATELIT**, direktori akan **disesuaikan menjadi**:  
```
../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../data/gambar
/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../data/gambar
/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg
```

Langkah ini bertujuan agar struktur folder lebih konsisten dan memudahkan proses pemrosesan citra selanjutnya.

In [133]:
# Membuat salinan dataframe
df_preprocessing_6 = df_preprocessing_5.copy()

# Menyesuaikan direktori gambar
df_preprocessing_6["dir_gambar_ori"] = df_preprocessing_6["dir_gambar_ori"].str.replace("../hasil/", "../data/gambar/")
df_preprocessing_6["dir_gambar_titik"] = df_preprocessing_6["dir_gambar_titik"].str.replace("../hasil/", "../data/gambar/")
df_preprocessing_6["dir_gambar_kotak_dan_yolo"] = df_preprocessing_6["dir_gambar_kotak_dan_yolo"].str.replace("../hasil/", "../data/gambar/")

# Cek hasil perubahan untuk 1 baris pertama
print(df_preprocessing_6['dir_gambar_ori'][0])
print(df_preprocessing_6['dir_gambar_titik'][0])
print(df_preprocessing_6['dir_gambar_kotak_dan_yolo'][0])
df_preprocessing_6.head()

../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg
../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg


nomor kode_sekolah                  nama_sekolah  \
0      1     10100139  SMA NEGERI 1 LEMBAH SEULAWAH   
1      2     10100170       SMA NEGERI MODAL BANGSA   
2      3     10100176            SMAS MALEM PUTRA 1   
3      4     10100177          SMA NEGERI 1 LEUPUNG   
4      5     10100179        SMA NEGERI 1 PULO ACEH   

                                                                                  alamat  \
0  JL. BANDA ACEH-MEDAN KM.57,5 Lamtamot Kec. Lembah Seulawah Kab. Aceh Besar Prov. Aceh   
1   JL. BANDARA SIM KM. 12,5 Cot Geundreut Kec. Blang Bintang Kab. Aceh Besar Prov. Aceh   
2            JL.SYEH ABDULLAH KAN-AN Leugeu Kec. Darul Imarah Kab. Aceh Besar Prov. Aceh   
3      JL. BANDA ACEH CALANG KM-18 DAYAH MAMPLAM Kec. Leupung Kab. Aceh Besar Prov. Aceh   
4                JL. RAJA KANDANG Alue Riyeung Kec. Pulo Aceh Kab. Aceh Besar Prov. Aceh   

  provinsi akreditasi       long       lat  \
0     Aceh          A  95.682881  5.370129   
1     Aceh          A  95.395695  5.508437   
2     Aceh          C  95.330857  5.493038   
3     Aceh          B  95.241723  5.435694   
4     Aceh          B  95.144949  5.615454   

                                                                   dir_gambar_ori  \
0  ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                   dir_gambar_titik  \
0  ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                       dir_gambar_kotak_dan_yolo  \
0  ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg   
1       ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_modal_bangsa.jpg   
2             ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/smp_malem_putra_1.jpg   
3                ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sman_1_leupung.jpg   
4               ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_1_pulo_aceh.jpg   

  keterangan  kabupaten_kota  
0     sukses  Kab Aceh Besar  
1     sukses  Kab Aceh Besar  
2     sukses  Kab Aceh Besar  
3     sukses  Kab Aceh Besar  
4     sukses  Kab Aceh Besar

## Pengecekkan Gambar

Sebelum menyimpan dataset, perlu dilakukan pengecekan apakah file gambar **benar-benar ada** di direktori yang telah disesuaikan.  
Hal ini penting untuk memastikan bahwa setiap entri pada kolom:

- `dir_gambar_ori`  
- `dir_gambar_titik`  
- `dir_gambar_kotak_dan_yolo`  

merujuk pada file gambar yang valid.

In [134]:
# Fungsi untuk cek keberadaan file
def cek_file(path):
    return os.path.exists(path)

# Membuat kolom baru untuk menandai apakah file ada atau tidak
df_preprocessing_6['ori_exists'] = df_preprocessing_6['dir_gambar_ori'].apply(cek_file)
df_preprocessing_6['titik_exists'] = df_preprocessing_6['dir_gambar_titik'].apply(cek_file)
df_preprocessing_6['kotak_yolo_exists'] = df_preprocessing_6['dir_gambar_kotak_dan_yolo'].apply(cek_file)

# Cek hasil, jumlah file yang tidak ditemukan
print("File ori tidak ada:", (~df_preprocessing_6['ori_exists']).sum())
print("File titik tidak ada:", (~df_preprocessing_6['titik_exists']).sum())
print("File kotak yolo tidak ada:", (~df_preprocessing_6['kotak_yolo_exists']).sum())


File ori tidak ada: 0
File titik tidak ada: 0
File kotak yolo tidak ada: 0


Setelah dicek, variabel `file_hilang` ternyata kosong.  
Artinya semua gambar tersedia di direktori masing-masing tanpa ada yang hilang.


---
# **EXPORT CLEANED DATASET**
---

In [143]:
# Membuat salinan dataframe hasil preprocessing
df_bersih = df_preprocessing_6.copy()
df_bersih = df_bersih[['nomor', 'kode_sekolah', 'nama_sekolah', 'alamat', 'kabupaten_kota', 'provinsi',
       'akreditasi', 'long', 'lat', 'dir_gambar_ori', 'dir_gambar_titik',
       'dir_gambar_kotak_dan_yolo', 'keterangan']]
print(df_bersih.shape)
df_bersih.head()

(14076, 13)


nomor kode_sekolah                  nama_sekolah  \
0      1     10100139  SMA NEGERI 1 LEMBAH SEULAWAH   
1      2     10100170       SMA NEGERI MODAL BANGSA   
2      3     10100176            SMAS MALEM PUTRA 1   
3      4     10100177          SMA NEGERI 1 LEUPUNG   
4      5     10100179        SMA NEGERI 1 PULO ACEH   

                                                                                  alamat  \
0  JL. BANDA ACEH-MEDAN KM.57,5 Lamtamot Kec. Lembah Seulawah Kab. Aceh Besar Prov. Aceh   
1   JL. BANDARA SIM KM. 12,5 Cot Geundreut Kec. Blang Bintang Kab. Aceh Besar Prov. Aceh   
2            JL.SYEH ABDULLAH KAN-AN Leugeu Kec. Darul Imarah Kab. Aceh Besar Prov. Aceh   
3      JL. BANDA ACEH CALANG KM-18 DAYAH MAMPLAM Kec. Leupung Kab. Aceh Besar Prov. Aceh   
4                JL. RAJA KANDANG Alue Riyeung Kec. Pulo Aceh Kab. Aceh Besar Prov. Aceh   

   kabupaten_kota provinsi akreditasi       long       lat  \
0  Kab Aceh Besar     Aceh          A  95.682881  5.370129   
1  Kab Aceh Besar     Aceh          A  95.395695  5.508437   
2  Kab Aceh Besar     Aceh          C  95.330857  5.493038   
3  Kab Aceh Besar     Aceh          B  95.241723  5.435694   
4  Kab Aceh Besar     Aceh          B  95.144949  5.615454   

                                                                   dir_gambar_ori  \
0  ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../data/gambar/gambar_ori/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                   dir_gambar_titik  \
0  ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_1_lembah_seulawah.jpg   
1       ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_negeri_modal_bangsa.jpg   
2             ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/smp_malem_putra_1.jpg   
3                ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sman_1_leupung.jpg   
4               ../data/gambar/gambar_titik/ACEH/KAB ACEH BESAR/sma_1_pulo_aceh.jpg   

                                                                       dir_gambar_kotak_dan_yolo  \
0  ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_1_lembah_seulawah.jpg   
1       ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_negeri_modal_bangsa.jpg   
2             ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/smp_malem_putra_1.jpg   
3                ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sman_1_leupung.jpg   
4               ../data/gambar/gambar_kotak_dan_yolo/ACEH/KAB ACEH BESAR/img/sma_1_pulo_aceh.jpg   

  keterangan  
0     sukses  
1     sukses  
2     sukses  
3     sukses  
4     sukses

Setelah semua tahapan preprocessing selesai, dataset kini sudah **bersih, lengkap, dan direktori gambar sudah disesuaikan** sebanyak **14.076 sekolah dengan 13 kolom**.  
Dataset ini akan disimpan ke file CSV pada direktori output:

`../data/data_preprocessing.csv`

File ini nantinya akan digunakan sebagai **input untuk proses selanjutnya** pada proyek **CLUSTERING-CITRA-SATELIT**.

In [144]:
# Menyimpan dataframe bersih ke file CSV di folder OUTPUT_DIR
df_bersih.to_csv(f"{OUTPUT_DIR}/data_preprocessing.csv", index=False)